WooCommerce/Classes/ViewRelated/Dashboard/Reviews/ReviewsDashboardCardViewModel.swift

```diff
@@ -19,11 +19,22 @@ final class ReviewsDashboardCardViewModel: ObservableObject {
         case approved
     }
 
-    @Published private(set) var data: [ReviewViewModel] = []
+    @Published private(set) var data: [ReviewViewModel] = [] {
+        didSet {
+            showLoadingAnimation = data.isEmpty
```

Comment:
Another way is keeping `showLoadingAnimation` a computed variable and `syncingData` as a private `@Published`. In that case, `showLoadingAnimation` will be updated when either `syncingData` or `data` is updated, which triggers the SwiftUI view to re-render.

Reply:
Thanks! This, the combine example works very nicely.

In my test, it also creates a side effect of not showing loading animation when switching status, if there's already local data for reviews with hold/approved status. This is expected since we want to immediately show local data, but in practice the switch requires a second or so to complete, so without loading animation the waiting looks a bit jarring.

As a solution, I'm adding a new variable `switchingStatus` and make the status text shimmer during switching action, video below:

https://github.com/woocommerce/woocommerce-ios/assets/266376/4c77ca51-598f-4771-b615-5a48e701426a

Reply:
Added combine and status switch animation on https://github.com/woocommerce/woocommerce-ios/pull/12769/commits/4dd707bc6c8afb34f0170d9728718bab94464309